File: capstone/java/src/main/java/com/techelevator/service/BreweryService.java
package com.techelevator.service;

import com.techelevator.model.BrewSearchDTO;
import com.techelevator.model.Brewery;

import java.security.Principal;
import java.util.List;

public interface BreweryService {
    Brewery getBreweryById(int breweryId);
    Brewery getRandomBrewery();
    List<Brewery> getAllBreweries();
    List<Brewery> getSavedBreweries(Principal principal);
    Brewery addBreweryToSaved(int breweryId, Principal principal);
    Brewery createBrewery(Brewery newBrewery, Principal principal);
    Brewery updateBrewery(Brewery updatedBrewery, Principal principal);
    void deleteSavedBrewery(int breweryId, Principal principal);
    void deleteBrewery(int breweryId, Principal principal);

//    List<Brewery> searchBreweries(BrewSearchDTO searchTerms);
}

File: capstone/vue/src/services/BreweryService.js
import axios from 'axios';

export default {
  getBreweries() {
    return axios.get('/breweries');
  },
  
  getBreweryById(breweryId) {
    return axios.get(`/breweries/${breweryId}`);
  },

  getMySavedBreweries() {
    return axios.get('/mybreweries');
  },
  
  getRandomBrewery() {
    return axios.get('/breweries/random')
  },

  createBrewery(brewery) {
    return axios.post('/breweries', brewery);
  },

  addSavedBrewery(breweryId) {
    return axios.post(`/mybreweries/${breweryId}`)
  },

  updateBrewery(brewery) {
    return axios.put(`/breweries/${brewery.breweryId}`, brewery)
  },

  deleteSavedBrewery(breweryId) {
    return axios.delete(`mybreweries/${breweryId}`)
  },

  deleteBrewery(breweryId) {
    return axios.delete(`breweries/${breweryId}`)
  }

  //add saved
  // delete saved
  

}